lib/AST/DiagnosticEngine.cpp

```diff
@@ -468,6 +468,13 @@ static void formatDiagnosticArgument(StringRef Modifier,
     
     // Strip extraneous parentheses; they add no value.
     auto type = Arg.getAsType()->getWithoutParens();
+    
+    // If a SyntaxSugarType has an unresolved type, its printed
+    // representation may be confusing in diagnostics. Desugar it for clarity.
+    if (type->hasSyntaxSugar() && type->hasUnresolvedType()) {
+      type = type->getCanonicalType();
```

Comment:
I don't think you want to drop all the way down to the canonical type, just one level of desugaring. Otherwise `[CInt: _]` will go to `Dictionary<Int32, _>`.